### PR TITLE
fix: address module imports that end with .go

### LINKED ISSUE #481: Packages with `.` in them are incorrectly imported in mocks
Description
-------------

When importing the library `github.com/nats-io/nats.go` and generating mock, the resulting import in the generated mock is missing the `nats.go` part (`github.com/nats-io/`).

Mockery Version
--------

`12.13.1`

Golang Version
--------------

`go1.18 darwin/arm64`

Installation Method
-------------------

- [ ] Binary Distribution
- [ ] Docker
- [x] brew
- [ ] go install
- [x] Other: Another colleague compiled from source and gets the same result

Steps to Reproduce
------------------

1. Add the following import to a file: `nats "github.com/nats-io/nats.go"`
2. Use the package in an interface
3. Generate mocks for said interface
4. Observe that the generated mock import is `nats "github.com/nats-io/"`

### Expected Behavior

I expect to the generated mock to import `nats "github.com/nats-io/nats.go"`

### Actual Behavior

The generated mock imports `nats "github.com/nats-io/"`

### PR DESCRIPTION
**Pull request by meshuga**
Description
-------------

Please include a summary of the changes and the related issue. Please also include relevant motivation and context.

- Fixes #481 #249 #419

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

Version of Golang used when building/testing:
---------------------------------------------

- [ ] 1.11
- [ ] 1.12
- [ ] 1.13
- [ ] 1.14
- [ ] 1.15
- [ ] 1.16
- [ ] 1.17
- [x] 1.18

How Has This Been Tested?
---------------------------

See `TestGetLocalizedPathVendored` and `TestGetLocalizedPathModule` in `generator_test.go`

Checklist
-----------

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: pkg/generator_test.go [hunk 1]
```diff
@@ -2433,6 +2433,26 @@ func NewMockRequesterGenerics[TAny interface{}, TComparable comparable, TSigned
 	s.checkGeneration("generic.go", "RequesterGenerics", true, "", expected)
 }
 
+func (s *GeneratorSuite) TestGetLocalizedPathVendored() {
+	generator := s.getGenerator(testFile, "Requester", false, "")
+	result := generator.getLocalizedPath(context.Background(), "vendor/github.com/nats-io/nats.go/js.go")
+
+	s.Equal(
+		"github.com/nats-io/nats.go", result,
```
Comment (LandonTClipp) [inline pkg/generator_test.go:2441]: If you're vendoring a directory don't you want the returned path to be on-disk path instead of the URL?

**Comment by LandonTClipp** [comment]
@meshuga thanks for the contribution. I have a couple of comments/questions. Also, I'm adding table-driven tests for `getLocalizedPath` in https://github.com/vektra/mockery/pull/488. Could you rebase off of this once I merge it and add your test there? Thanks!

File: pkg/generator.go [hunk 2]
```diff
@@ -194,7 +194,7 @@ func (g *Generator) getLocalizedPath(ctx context.Context, path string) string {
 	log := zerolog.Ctx(ctx).With().Str(logging.LogKeyPath, path).Logger()
 	ctx = log.WithContext(ctx)
 
-	if strings.HasSuffix(path, ".go") {
+	if strings.HasSuffix(path, ".go") && strings.Contains(path, "vendor"+string(filepath.Separator)) {
```
[the remark this answers is not in the document]
Comment (meshuga) [reply pkg/generator.go:197]: Good point! I think we might break the logic in this case. It looks like I'd need to refactor the logic to ensure the path is only a module and nothing else.

File: pkg/generator_test.go (hunk 1 above)
Comment (meshuga) [reply pkg/generator_test.go:2441]: I'm golang code, you put imports to a vendored directory in same way as from modules, that's why this current logic is in place.

Comment (LandonTClipp) [reply pkg/generator_test.go:2441]: I see, I've never personally used vendored dependencies.

Comment (meshuga) [reply pkg/generator_test.go:2441]: Yeah, I used to. It's just a directory with all project dependencies in a form of a source code. In any case, we ought to check if a path element is a directory or not and if it's a file, remove it from path to build correct import statement.

**Comment by meshuga** [comment]
@LandonTClipp I think you merged the PR too early, The PR didn't have any changes. I was digging a little bit deeper in the code to understand how imports are generated, taking more time than expected :)

**Comment by LandonTClipp** [comment]
That's really odd I didn't intend to merge this. I must have fat fingered something! Sorry about that.

**Pull request merged**